src/sc/sidechaintypes.cpp

```diff
@@ -0,0 +1,373 @@
+    }
+    auto digest = ZendooPoseidonHash();
+
+    field_t* lhsFe = zendoo_deserialize_field(&(*lhs.byteVector.begin()));
+    if (lhsFe == nullptr) {
+        LogPrintf("%s():%d - failed to deserialize: %s \n", __func__, __LINE__, libzendoomc::ToString(zendoo_get_last_error()));
+        zendoo_clear_error();
+        throw std::runtime_error("Could not compute poseidon hash");
+    }
+    digest.update(lhsFe);
+
+    field_t* rhsFe = zendoo_deserialize_field(&(*rhs.byteVector.begin()));
+    if (rhsFe == nullptr) {
+        LogPrintf("%s():%d - failed to deserialize: %s \n", __func__, __LINE__, libzendoomc::ToString(zendoo_get_last_error()));
+        zendoo_clear_error();
+        zendoo_field_free(lhsFe);
+        throw std::runtime_error("Could not compute poseidon hash");
+    }
+    digest.update(rhsFe);
+
+    field_t* outFe = digest.finalize();
+    CFieldElement res;
+    res.byteVector.resize(CFieldElement::ByteSize());
+    zendoo_serialize_field(outFe, &*(res.byteVector.begin()));
+
+    zendoo_field_free(lhsFe);
+    zendoo_field_free(rhsFe);
+    zendoo_field_free(outFe);
+    return res;
+}
+#endif
+////////////////////////////// End of Field types //////////////////////////////
+
+/////////////////////// libzendoomc namespace definitions //////////////////////
+namespace libzendoomc
+{
+
+    bool IsValidScProof(const ScProof& scProof)
+    {
+        auto scProofDeserialized = zendoo_deserialize_sc_proof(scProof.begin());
+        if (scProofDeserialized == nullptr)
+            return false;
+        zendoo_sc_proof_free(scProofDeserialized);
+        return true;
+    }
+
+    bool IsValidScVk(const ScVk& scVk)
+    {
+        auto scVkDeserialized = zendoo_deserialize_sc_vk(scVk.begin());
+        if (scVkDeserialized == nullptr)
+            return false;
+        zendoo_sc_vk_free(scVkDeserialized);
+        return true;
+    }
+
+    std::string ToString(Error err){
+        return strprintf(
+            "%s: [%d - %s]\n",
+            err.msg,
+            err.category,
+            zendoo_get_category_name(err.category));
+    }
+}
+/////////////////// end of libzendoomc namespace definitions ///////////////////
+
+////////////////////////////// Custom Config types //////////////////////////////
+bool FieldElementCertificateFieldConfig::IsValid() const
+{
+    if(nBits > 0 && nBits <= SC_FIELD_SIZE*8)
+        return true;
+    else
+        return false;
+}
+
+FieldElementCertificateFieldConfig::FieldElementCertificateFieldConfig(int32_t nBitsIn):
+    CustomCertificateFieldConfig(), nBits(nBitsIn) {}
+
+int32_t FieldElementCertificateFieldConfig::getBitSize() const
+{
+    return nBits;
+}
+
+//----------------------------------------------------------------------------------
+bool BitVectorCertificateFieldConfig::IsValid() const
+{
+    bool isBitVectorSizeValid = (bitVectorSizeBits > 0) && (bitVectorSizeBits <= MAX_BIT_VECTOR_SIZE_BITS);
```

Comment:
verify also that bitVectorSizeBits is both modulus 254 and 8 (see comment on sidechaintypes.h)

Reply:
Done